### DISCUSSION TITLE
Experience on Raspberry Pi 4 model B and Raspbian OS

### DISCUSSION THREAD

**Discussion by kevpatts** [ct-Open-Source/tuya-convert#1110] [General]
Ensure you do in advance:
`sudo rpi-update`
update to the latest firmware
`sudo raspi-config`
ensure that Localisation Options => WLAN Country is set to US (AP mode vtrust-flash network doesn't show up without this).
`sudo reboot`
Once rebooted do the git clone command and CD into the directory. No need to change the config file as Raspi already uses wlan0.

If you run install_prereq.sh you'll get this error:
`error: externally-managed-environment

× This environment is externally managed
╰─> To install Python packages system-wide, try apt install
    python3-xyz, where xyz is the package you are trying to
    install.
    
    If you wish to install a non-Debian-packaged Python package,
    create a virtual environment using python3 -m venv path/to/venv.
    Then use path/to/venv/bin/python and path/to/venv/bin/pip. Make
    sure you have python3-full installed.
    
    For more information visit http://rptl.io/venv

note: If you believe this is a mistake, please contact your Python installation or OS distribution provider. You can override this, at the risk of breaking your Python installation or OS, by passing --break-system-packages.
hint: See PEP 668 for the detailed specification.`

To fix this you need to configure the python environment:
`python3 -m venv venv`
`source venv/bin/activate`
`sed -i "s/sudo -H python3 -m //" install_prereq.sh`
`sed -i "s@ -m -d \.@ -m -d ~/tuya-convert/venv/bin/python3 .@" start_flash.sh`
`sed -i "s@~/tuya-convert/venv/bin/python3 ./setup_ap.sh@./setup_ap.sh@" start_flash.sh`
Then run install_prereq.sh again.
In future before you run the script you have to change to the correct python environment by running
`source venv/bin/activate` in the tuya-convert directory.

You HAVE TO DISABLE Wireless LAN using the icon in the top right. Otherwise this will not work.
Everything SHOULD run normally then. Don't know how easily it is to update the repo to account for these, but there ya go.

**Comment by OracPrime** [comment]
This needs to be in the readme!

"ensure that Localisation Options => WLAN Country is set to US (AP mode vtrust-flash network doesn't show up without this)."

But thank you @kevpatts  for mentioning it here, I was tearing my hair out.
